src/kernel/arch/x86/paging.zig

```diff
@@ -94,15 +149,91 @@ fn mapDir(dir: *Directory, phys_start: u32, phys_end: u32, virt_start: u32, virt
         return PagingError.UnalignedVirtAddresses;
     }
 
+    const entry = virt_start / PAGE_SIZE_4MB;
+    var dir_entry = &dir.entries[entry];
```

Comment:
Idk whether it is worth having a check out of bounds for the array

Reply:
It is impossible for it to be out of bounds (because virt_start is a usize and a 32 bit int divided by 4MB will be less than 1024) but will add one for the sake of completion.